pystac/cache.py

```diff
@@ -20,12 +25,12 @@ def get_cache_key(stac_object):
     if href is not None:
```

Comment:
Prefer `if href:`.  When can a return from get_self_href() eval to None and not be None?